flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CompletedCheckpoint.java

```diff
@@ -281,6 +285,30 @@ private void doDiscard() throws Exception {
 		}
 	}
 
+	private Set<StreamStateHandle> getSharedStateHandle() {
+		Set<StreamStateHandle> sharedStateHandles = new HashSet<>(operatorStates.size());
+
+		//reference SavepointV2Serializer.serializeKeyedStateHandle
+		operatorStates.values().forEach(operatorState -> {
+			operatorState.getStates().forEach(operatorSubtaskState -> {
+				if (operatorSubtaskState != null) {
+					for (KeyedStateHandle managedKeyedState : operatorSubtaskState.getManagedKeyedState()) {
+						if (managedKeyedState instanceof IncrementalRemoteKeyedStateHandle) {
+							sharedStateHandles.addAll(((IncrementalRemoteKeyedStateHandle) managedKeyedState).getSharedState().values());
```

Comment:
Cleaning up all shared state(FileStateHandle) will cause SharedStateRegistry to become invalid, and all shared state will be cleaned up?

Please confirm, has the sst file used by subsequent Checkpoint been deleted?